HealthcareBase/Service/ScheduleService/ProcedureService/DoctorCredentialsDto.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using HealthcareBase.Model.Filters;
+using Model.Schedule.Procedures;
+using Model.Users.Employee;
+
+namespace Service.ScheduleService.ProcedureService
+{
+    public class DoctorCredentialsDto : AbstractFilter<Examination, int>
+    {
+        public string Name { get; set; }
+        public string Surname { get; set; }
+
+        protected override void ConfigureFilter()
+        {
+            if (!string.IsNullOrEmpty(Name)) 
+                AddExpressionFunction(examination => examination.Doctor.Person.Name.Contains(Name));
+            if (!string.IsNullOrEmpty(Surname))
+                AddExpressionFunction(examination => examination.Doctor.Person.Surname.Contains(Surname));
+        }
```

Comment:
Violated Law of Demeter - "use only one dot"

Reply:
Advanced query requires access to doctor's surname, therefore it must be accessed this way.